lib/reek/report/report.rb

```diff
@@ -121,6 +123,41 @@ def total_smell_count_message
     end
 
     #
+    # Displays the status of each file as it is examined
+    # Displays the text report when complete
+    #
+    # @public
+    class ProgressReport < TextReport
```

Comment:
There is a lot of duplication between this and the TextReport class, so it felt cleaner to subclass instead of creating an entirely independent class.

It's not clear to me how options would be passed down to the Report class, so if I was to re-implement this I could rename the class to be `TextReportWithProgress` and still subclass the `TextReport`. Then I would use the options to decide if we call `TextReport` or `TextReportWithProgress`.

Do you think the duplication would be worth it?